src/script/sign.h

```diff
@@ -572,9 +576,12 @@ struct PartiallySignedTransaction
     bool IsNull() const;
     void Merge(const PartiallySignedTransaction& psbt);
     bool IsSane() const;
+    bool AddInput(const CTxIn& txin, PSBTInput& psbtin);
+    bool AddOutput(const CTxOut& txout, const PSBTOutput& psbtout);
     PartiallySignedTransaction() {}
     PartiallySignedTransaction(const PartiallySignedTransaction& psbt_in) : tx(psbt_in.tx), inputs(psbt_in.inputs), outputs(psbt_in.outputs), unknown(psbt_in.unknown) {}
     explicit PartiallySignedTransaction(const CMutableTransaction& tx);
+    bool GetInputUTXO(CTxOut& utxo, int input_index) const;
```

Comment:
Please add a short doxygen documentation for the new calls.
(I know, the current ones don't, but every day is a good day to get started)

Reply:
Maybe we can avoid the input/output order completely by returning an `Optional<CTxOut>` instead. We have this in `optional.h`, after all.

Reply:
Added doxygen comment